pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/FunctionRuntimeManager.java

```diff
@@ -151,7 +151,7 @@ public FunctionRuntimeManager(WorkerConfig workerConfig, WorkerService workerSer
         }
         secretsProviderConfigurator.init(workerConfig.getSecretsProviderConfiguratorConfig());
 
-        Optional<FunctionAuthProvider> functionAuthProvider = Optional.empty();
```

Comment:
Why do we need to change this?

Reply:
Because we now allow namespaces to be overridden by the ManifestCustomizer, we end up with an authprovider class being created per function, as the authProvider might be writing creds to different namespaces depending on the function. We do that by delaying the instantiation of the AuthProvider until we have the function details, so we just pass the classname into runtime providers.